jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CloudFormation.kt

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation
+
+import software.amazon.awssdk.services.cloudformation.CloudFormationClient
+import software.amazon.awssdk.services.cloudformation.model.CloudFormationException
+import software.amazon.awssdk.services.cloudformation.model.DescribeStacksRequest
+import software.amazon.awssdk.services.cloudformation.model.StackStatus
+import software.aws.toolkits.core.utils.wait
+import software.aws.toolkits.resources.message
+import java.time.Duration
+
+fun CloudFormationClient.executeChangeSetAndWait(stackName: String, changeSet: String) {
+    val isCreate = try {
+        val describeStack = this.describeStacks { it.stackName(stackName) }
+        describeStack.stacks().firstOrNull()?.stackStatus() == StackStatus.REVIEW_IN_PROGRESS
+    } catch (e: CloudFormationException) {
+        if (e.message?.contains("Stack with id $stackName does not exist") == true) {
```

Comment:
Is it better to use `e.awsErrorDetails().errorCode() == "ValidationError"`?
This seems hacky to use the error message which is likely to change over time.